### ISSUE TITLE
Enforce correct syntax in relative imports

### ISSUE THREAD

**Issue by yguclu** [pyccel/pyccel#294]
According to the [official documentation](https://docs.python.org/3/reference/import.html#package-relative-imports) a "relative import" statement must start with `from`, and not with `import`. Instead Pyccel currently accepts the wrong syntax
```
import .module as mod
```
and interprets it as the correct form
```
from . import module as mod
```
If we move away from [redbaron](https://redbaron.readthedocs.io/en/latest/) and use the standard [ast](https://docs.python.org/3/library/ast.html) module instead, Pyccel will automatically be able to identify any syntax errors. Therefore I do not think that we should fix this problem now, and I am giving it a **wontfix** label.

@saidctb @EmilyBourne: any comments about this?

**Comment by EmilyBourne** [comment]
I don't think it is worth fixing for now. I'm not sure it is really a problem if we support more than we are supposed to and the behaviour is what would be expected if this syntax did exist.

We don't currently check that the python syntax is correct at all. I think it is entirely possible that you can get some strange results by passing invalid python code to pyccel but up until now we have said that it is the user's job to ensure that they write valid python code before they call pyccel

**Comment by yguclu** [comment]
I have written this issue because I have encountered a situation where a Python project accelerated with Pyccel could not run in pure Python after refactoring, but it could still run using Pyccel. The developer did not realize that this was due to wrong import syntax. A situation like this could potentially lead to undefined behavior if the development proceeds without fixing the underlying error.

I think redbaron should be able to check the syntax for us, but apparently there are some issues there.
